agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2666,6 +2669,54 @@ bool backhaul_manager::handle_1905_combined_infrastructure_metrics(
     return send_cmdu_to_bus(cmdu_tx, src_mac, bridge_info.mac);
 }
 
+bool backhaul_manager::handle_1905_topology_discovery(const std::string &src_mac,
+                                                      ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto tlvAlMac = cmdu_rx.getClass<ieee1905_1::tlvAlMacAddressType>();
+    if (!tlvAlMac) {
+        LOG(ERROR) << "addClass tlvAlMacAddressType failed";
+        return false;
+    }
+
+    // Filter out the messages we have sent.
+    if (tlvAlMac->mac() == network_utils::mac_from_string(bridge_info.mac)) {
+        return true;
+    }
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received TOPOLOGY_DISCOVERY_MESSAGE from AL MAC=" << tlvAlMac->mac()
+              << ", mid=" << std::hex << int(mid);
+
+    auto new_device = m_1905_neighbor_devices.find(tlvAlMac->mac()) == m_1905_neighbor_devices.end()
+                          ? true
+                          : false;
```

Comment:
This is redundant, just use
```suggestion
auto new_device = m_1905_neighbor_devices.find(tlvAlMac->mac()) == m_1905_neighbor_devices.end();
```

Reply:
done.

Reply:
Looks like this was missed.

Reply:
it is on a fixup
https://github.com/prplfoundation/prplMesh/pull/1019/commits/e55c5493dbb21b8330b272192c8d0d6e21ec6363